### ISSUE TITLE
Plans to provide this utility as a library to be used within other projects

### ISSUE THREAD

**Issue by kgritesh** [segmentio/golines#34]
This is a fantastic utility - kudos for taking the effort to build it. I was wondering if you would be open to the idea of providing this utility also as a library so that any other projects working with code generation can use this to format the generated code. 

My use case is that I work on a no-code project where we generate a lot of go code for our customers. The generated code has a huge map which by default is rendered on the same line and is impossible to grok with naked eyes. So I wanted to introduce golines as one of the preprocessing step to break down the map into individual lines. Hence if this utility is available as a library- it would be super useful for me. 

I am happy to provide a PR for this - but will require some refactoring and hence wanted to discuss before working on it.

**Comment by yolken-segment** [comment]
Sure, feel free to submit a PR for this.

**Comment by kgritesh** [comment]
@yolken-segment Yes i saw the code and that's what it will take. I will try to work on it as time permits and submit a PR. Thanks for this library and the accompanying blog post.

**Comment by strideynet** [comment]
@kgritesh How is this coming along ? I'm happy to raise a PR for this sometime tomorrow if you aren't interested in doing it still ?
